CanHazFunny/CanHazFunny.Tests/JesterTests.cs

```diff
@@ -1,7 +1,74 @@
-using Xunit;
-
-namespace CanHazFunny.Tests;
-
-public class JesterTests
-{
-}
+using Xunit;
+using Moq;
+using System;
+
+namespace CanHazFunny.Tests;
+
+public class JesterTests
+{
+    [Fact]
+    public void Jester_JokeServiceIsNull_ThrowException()
+    {  
+        //Arrange
+        IJokeServiceOutput jokeOutput = new JokeServiceOutput();
+
+        // Act and Assert
+        Assert.Throws<ArgumentNullException>(() => new Jester(null!, jokeOutput));
+
+    }
+
+    [Fact]
+    public void Jester_JokeOutputIsNull_ThrowException()
+    {
+        //Arrange
+        IJokeService jokeService = new JokeService();
+
+        //Act and Assert
+        Assert.Throws<ArgumentNullException>(() => new Jester(jokeService, null!));
+    }
+
+    [Fact]
+    public void TellJoke_ReturnsJoke_Success()
+    {
+        // Arrange
+        var jokeServiceMock = new Mock<IJokeService>();
+        var outputInterfaceMock = new Mock<IJokeServiceOutput>();
+
+        //Return two jokes, neither contain Chuck Norris
+        jokeServiceMock.SetupSequence(s => s.GetJoke())
+            .Returns("Why did the chicken cross the road? To get to the other side!")
+            .Returns("This is another joke");
+
+        var jester = new Jester(jokeServiceMock.Object, outputInterfaceMock.Object);
+
+        // Act
+        jester.TellJoke();
+
+        // Assert
+        outputInterfaceMock.Verify(o => o.Write("Why did the chicken cross the road? To get to the other side!"), Times.Once);

```

Comment:
Test looks good but not sure if you want to convert to the Assert.Equal() format for this test.
```suggestion
        Assert.Equal("Normal joke", mockJokeOutput.Invocations[0].Arguments[0] as string);

```

Reply:
As per other comment, kept outputInterfaceMock.Verify but also added Assert.Equal check after